recipe/thirdparty/tpmodels/providers.go

```diff
@@ -15,66 +15,147 @@
 
 package tpmodels
 
+import (
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+type TypeGoogleInput struct {
+	Config   []GoogleConfig
+	Override func(provider *GoogleProvider) *GoogleProvider
+}
+
 type GoogleConfig struct {
-	ClientID              string
-	ClientSecret          string
-	Scope                 []string
-	AuthorisationRedirect *struct {
-		Params map[string]interface{}
-	}
-	IsDefault bool
+	ClientID     string
+	ClientSecret string
```

Comment:
Sometimes client secret is not available for google - for example, in the case of mobile app flows, google does not give you a client secret since it expects you to follow the frontend PKCE flow. So this should be optional (and also in other providers, we should check in which it may not be there)

Reply:
Having pointer to string kind of makes it inconvenient to configure the value. For primitive fields, we can always skip defining them, which gives a sense that it's optional. Something like:
```go
tpmodels.GoogleConfig{
  ClientID: "client-id",
}
```

We can just check for empty string internally.